packages/plugin-commands-publishing/src/publish.ts

```diff
@@ -102,18 +102,19 @@ export async function handler (
         hint: GIT_CHECKS_HINT,
       })
     }
-    const branch = opts.publishBranch ?? (await getDefaultBranch() || 'main')
+    const configuredBranch = opts.publishBranch ?? await getDefaultBranch()
+    const branches = configuredBranch ? [configuredBranch] : ['master', 'main']
     const currentBranch = await getCurrentBranch()
-    if (currentBranch !== branch) {
+    if (branches.includes(currentBranch)) {
       const { confirm } = await prompt({
-        message: `You're on branch "${currentBranch}" but your "publish-branch" is set to "${branch}". \
+        message: `You're on branch "${currentBranch}" but your "publish-branch" is set to "${branches.join('/')}". \
```

Comment:
a branch name may contain `/`, so to avoid confusion, let's probably use `|` to join the branch names. And in the error as well